src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        result = 31 * result + ((width == null) ? 0 : width.hashCode());
+        result = 31 * result + ((length == null) ? 0 : length.hashCode());
+        result = 31 * result + ((color == null) ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || !object.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        return (width == null ? rectangle.width == null : width == rectangle.width)
```

Comment:
Integer a = 128;
        Integer b = 128;
       boolean result = a == b; 
result will be true or false? and why(answer in comment below)